lib/CL/pocl_util.c

```diff
@@ -64,6 +75,126 @@ typedef struct list_item
 } list_item;
 
 
+#ifdef __linux__
+#ifdef __x86_64__
```

Comment:
I have tested after the first commit in this PR (-cl-ext) and it did compile and run some tests (from -L internal), but not all work.